src/libraries/Common/src/System/Collections/Generic/ReferenceEqualityComparer.cs

```diff
@@ -15,7 +16,7 @@ private ReferenceEqualityComparer()
         {
         }
 
-        public bool Equals(T x, T y)
+        public bool Equals([AllowNull] T x, [AllowNull] T y)
```

Comment:
Since `T` is constraint to class, can we instead annotate `T` as `T?` ?

Reply:
Indeed. I'd missed that. Thanks

Reply:
Hum, it looks like this file is compiled into more than one project. One with nullability enabled and one without. Reverted to use the attribute.

Reply:
Can you use the annotation and just add `#nullable enable` at the top of the file as well?  That's what we've done in other cases like this, where we've not yet gotten to every project that includes it.